include/seastar/core/sstring.hh

```diff
@@ -673,6 +673,23 @@ operator<<(std::basic_ostream<char_type, char_traits>& os,
     return os.write(s.begin(), s.size());
 }
 
+#if __cpp_lib_three_way_comparison
+
+template <typename char_type, typename size_type, size_type max_size>
+std::strong_ordering
+operator<=>(const char_type* s1, const basic_sstring<char_type, size_type, max_size>& s2) {
+    return std::basic_string_view<char_type>(s1) <=> std::basic_string_view<char_type>(s2);
+}
+
+template <typename string_view_like, typename char_type, typename size_type, size_type max_size>
+SEASTAR_CONCEPT(requires std::convertible_to<std::basic_string_view<char_type>, string_view_like>)
+std::strong_ordering
+operator<=>(const string_view_like& s1, const basic_sstring<char_type, size_type, max_size>& s2) {
+    return std::basic_string_view<char_type>(s1) <=> std::basic_string_view<char_type>(s2);
+}
```

Comment:
if we are using the <=> operator to implement the comparison operators, then we might want to implement the `operator==` as well.

Reply:
Can't operator== be synthesized from operator<=>?

Reply:
Not exactly: if `<=>` is _defaulted_, then == is also defaulted (if it wasn't user-specified). However if you provide a `<=>` operator, neither `==` nor `!=` will be synthesized using it. So in neither case is (in-)equality implemented with spaceship: they are two separate domains.

Reply:
yeah. the defaulted one and the non-default one are different in this case.